### PR TITLE
Add toggle to enable/disable agent selection - default is agent selection is off

### PR DESCRIPTION
**Pull request by mamoodi**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**
There is a lot of confusion on what agent to use when using OpenDevin. Most if not all users should just use the CodeActAgent for all their tasks. 

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**
Add a toggle that enables the agent selection but beside it say "Advanced Users" so people are less likely to change it.
The agent selection is always disabled when you open the settings UI.

Please do not auto-merge this. I want a few maintainers to have a look since this is user facing changes.

---
**Other references**
Screenshot 2024-07-30 at 7 31 36 PM

### CODE REVIEW

**Comment by amanape** [comment]
Are we planning on removing the ability to choose permanently? If so, you should just remove the combobox and all references to agent entirely rather than add a hidden prop.

If it is temporary, and you don't want to use agent-related components then you can simply do something like this:

```tsx
    <>
      {false && (
        <AutocompleteCombobox
          ariaLabel="agent"
          items={agents.map((agent) => ({ value: agent, label: agent }))}
          defaultKey={settings.AGENT}
          onChange={onAgentChange}
          tooltip={t(I18nKey.SETTINGS$AGENT_TOOLTIP)}
          disabled={disabled}
        />
      )}
    <>
```

**Comment by enyst** [comment]
I think it's temporary. @mamoodi correct me if I'm missing something, I guess it was due to the many micro-agents that aren't that user-friendly and it confuses people. When we have two or more agents easy to use for people, it makes sense to show the choice.

Personally, I'd prefer a setting in the UI settings to re-enable this at will.

**Comment by mamoodi** [comment]
@enyst yes you are correct. The current list of agents are mostly used for testing, they don't really have a use case. We want OpenDevin to be more user friendly especially when things things start up and the user is getting familiar with the application.

The goal in the future is to hopefully provide a very select number of agents that actually have proper use cases.

I can attempt a solution that hides the dropdown at will but I'm no UI designer or UI dev heh.

**Comment by amanape** [comment]
In this case, I think a good alternative would be adding a (!) above the select to let the user know this is an advanced use case setting and they most likely just want the default setting. If we have an agent reference somewhere in docs, we could reference it too. For example:

```
<------------------------------------------------>
      (!) advanced: You most likely don't need to 
           change this and can leave it as default.
           For more info on agents, read here.
      <---------------------->
         agent: <CodeActAgent>
      <---------------------->

      <---------------------->
         model: <gpt-4o>
      <---------------------->

      // other options ...
<------------------------------------------------>
```

I prefer making a change like this or a toggle that locks/unlock this change as another way to let the user know they shouldn't really change this. I don't think it is a good idea to hide this option right now.

Plus, on the off chance there are or have been users that created and/or use another agent, now they no longer have a way to use or select it.

What do you guys think?

**Comment by mamoodi** [comment]
@amanape that makes sense. I can try to work on this on the side. See if I can get something working. Shouldn't be too bad.
If I can't get the hide toggle, I'll add a message like you suggested

**Comment by mamoodi** [comment]
Closing. Will reopen when I have something ready.

**Comment by mamoodi** [comment]
@amanape I gave this another shot. Let me know what you think. Updated description with new changes and a screenshot of what it looks like.

**Review by amanape — APPROVED** [review]

**Comment by tobitege** [comment]
Nice!

File: frontend/src/components/modals/settings/SettingsForm.test.tsx [hunk 1]
```diff
@@ -119,17 +119,18 @@ describe("SettingsForm", () => {
     });
 
     it("should call the onAgentChange handler when the agent changes", async () => {
+      const user = userEvent.setup();
       renderSettingsForm();
 
+      // We need to enable the agent select first
```
Comment (amanape) [inline frontend/src/components/modals/settings/SettingsForm.test.tsx:125]:
```suggestion
      // We need to enable the agent select
```

**Pull request merged**